packages/console-extension/src/index.ts

```diff
@@ -276,6 +294,68 @@ async function activateConsole(
   const sessionDialogs =
     sessionDialogs_ ?? new SessionContextDialogs({ translator });
 
+  const pluginId = '@jupyterlab/console-extension:tracker';
+  const promptCellPositions: CodeConsole.PromptCellPosition[] = [
+    'top',
+    'bottom',
+    'left',
+    'right'
+  ];
+
+  // Instantiate the toolbar factory for console panel at plugin activation time
+  // since the plugin defines toolbar items and "jupyter.lab.toolbars" is set to true.
+  let toolbarFactory: ReturnType<typeof createToolbarFactory> | undefined;
+  if (toolbarRegistry) {
+    const factory = 'ConsolePanel';
+    toolbarFactory = createToolbarFactory(
+      toolbarRegistry,
+      settingRegistry,
+      factory,
+      pluginId,
+      translator
+    );
+
+    if (sessionContextDialogs) {
+      toolbarRegistry.addFactory<ConsolePanel>(factory, 'kernelName', panel =>
+        Toolbar.createKernelNameItem(
+          panel.sessionContext,
+          sessionContextDialogs,
+          translator
+        )
+      );
+    }
+
+    toolbarRegistry.addFactory<ConsolePanel>(factory, 'kernelStatus', panel => {
+      const sessionContext = panel.sessionContext;
+      // TODO: update use of deprecated APIs, without having to depend on @jupyterlab/notebook
+      const indicator = Toolbar.createKernelStatusItem(sessionContext);
+      return indicator;
+    });
+
+    const promptMenu = new Menu({ commands });
+    promptMenu.addClass('jp-CodeConsolePromptMenu');
+    promptCellPositions.forEach(position => {
+      promptMenu.addItem({ command: `console:prompt-to-${position}` });
+    });
+
+    toolbarRegistry.addFactory<ConsolePanel>(
+      factory,
+      'promptPosition',
+      panel => {
+        const button = new ToolbarButton({
+          tooltip: trans.__('Change Console Prompt Position'),
+          icon: dotsIcon,
+          onClick: () => {
+            const left = button.node.getBoundingClientRect().left;
+            const bottom = button.node.getBoundingClientRect().bottom;
+            promptMenu.open(left, bottom);
```

Comment:
IMO we will want to set `horizontalAlignment` here (but we have not released https://github.com/jupyterlab/lumino/pull/732 yet). I opened https://github.com/jupyterlab/jupyterlab/issues/17213 to tackle it in a follow-up PR